anakondo.el

```diff
@@ -197,17 +202,28 @@ and the completion candidates for it at cdr for buffer.")
 
 We try to find the project root by:
 1. Trying to query `clojure-mode' for it.
-2. Trying to query projectile for it.
-3. Defaulting to the `default-directory' of the buffer otherwise.
+2. Trying to query `project' for it.
+3. Trying to query projectile for it.
+4. Defaulting to the `default-directory' of the buffer otherwise.
 
 Anaphoric macro, binds `root' implicitly."
-  `(let* ((root (or (clojure-project-dir)
-                    (projectile-project-root)
+  `(let* ((root (or (and (featurep 'clojure-mode) (clojure-project-dir))
+                    (anakondo--project-get-project-root)
+                    (and (featurep 'projectile) (projectile-project-root))
                     default-directory)))
      ,@body))
 
 ;;;; Functions
 
+(defun anakondo--project-get-project-root ()
+  "Wrapper around `project.el' for getting project root."
+  (when-let ((project (and (featurep 'project)
+                           (project-current))))
+    (if (fboundp #'project-root)
+        (project-root project)
+      (car (with-no-warnings
+             (project-roots project))))))
```

Comment:
Can you explain why `project-root` might be missing? And than you'd instead return the last project of `project-roots`? Was it that `project-root` was added later in `project.el`?

A comment in the code about this would be nice.

Reply:
This logic is inspired by how LSP mode checks project root via project.el. As for why, I've found this: https://code.librehq.com/kfogel/emacs/-/commit/5044c19001fe608f2eac621add2e05cbca6c804b

Basically project-root is new function and may not be available, but old one returned several roots. I was safely using `car`ring from `project-roots` result for some years, but as `project-roots` now deprecated, `project-root` should be used instead.

Reply:
Sounds good, think you summarize this as a comment in the code as well? I know future me will have the same question over and over for this code.